### PR TITLE
#2927: Blocked from starting a new domain request when there are other started domain requests - [MEOWARD]

### LINKED ISSUE #2927: Blocked from starting a new domain request when there are other started domain requests
### Current Behavior

When I click start a new domain request I am brought to a request I've already started

### Expected Behavior

When I click start a new domain request I am brought to a brand new form with nothing filled in 

### Steps to Reproduce

1. login to a non-org account
2. Start a domain request, fill in the first 3 pages
3. Go back and start another new request

### Environment

Firefox

### Additional Context

https://github.com/user-attachments/assets/059404e4-c83b-40cc-ab90-c47572f71a03

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #2927

## Changes

- Changes the DomainRequestWizard such that the domain request flow now takes an id in the url path, rather than in the session
- Fixes existing tests on the domain request page to pass in an id
- Adds a breadcrumb to the first page of the domain request flow (such that you can go back to the home page), and kept the old back button on the rest

## Context for reviewers
**Context for design:**
The only design change is a breadcrumb on the first page of the domain request flow. This exists on the non-org view as well.

**General context:**
This bug was happening because the original DomainRequestWizard was actually storing the domain request id and the "new_request" variable in the session. Because our session is in our database as a CacheTable, this meant that you could override the session state by taking other actions in another tab.

The fix for this is to simply expose the domain request id in the actual url itself, rather than relying on session to store this information for us. 

Because we are now removing the id, this does reintroduce the [back button bug](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1730749487710769). It was decided that this would be resolved with a new design, so a breadcrumb has also been added to the first step of the domain request flow.

## Setup

**(devs only) You can reproduce the original bug on another sandbox by doing the following:**
1. Open the app in two tabs (or two browsers)
2. On one of the tabs, start a new domain request and complete some of it. Stop midway through.
3. On the other tab, click start a new domain request again. Either one of two things will happen, depending on the _order_ in which you complete these steps: the request will look like its already filled out, _or_ the request on your original tab will lose its progress (you will need to click on another step to trigger another get or post) 

For this ticket, follow these same steps on getgov-meoward. This behavior should now no longer exist.

**To test the new breadcrumb (both portfolio and non portfolio:**
1. Start a new domain request, and note the breadcrumb on the first step
2. Go to the second step. Note the old context.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [x] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [x] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Verified code meets all checks above. Address any checks that are not satisfied
- [x] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [x] Checked that all code is adequately covered by tests
- [x] Verify migrations are valid and do not conflict with existing migrations - NA

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [x] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Meets all designs and user flows provided by design/product
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] (Rarely needed) Tested as both an analyst and applicant user - NA

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

## Screenshots
image

### CODE REVIEW

**Comment by erinysong** [comment]
Tested the following:
1. Created a new domain request, filled form up to senior official and exited back to home page. image
2. Click on create a domain request button and was directed to a blank new form with a new session id 
image
3. When I go back to my original incompleted domain request in step 1, I am redirected back to where I last left the domain request form 
image

File: src/registrar/views/domain_request.py [hunk 1]
```diff
@@ -182,7 +183,9 @@ def configure_step_options(self):
 
     def has_pk(self):
         """Does this wizard know about a DomainRequest database record?"""
-        return "domain_request_id" in self.storage
+        if self.kwargs.get("id") is not None:
```
Comment (erinysong) [inline src/registrar/views/domain_request.py:186]: does this work the same if we one liner this to `return self.kwargs.get("id")` or `return self.kwargs.get("id") is not None`?

File: src/registrar/views/domain_request.py [hunk 2]
```diff
@@ -493,6 +493,7 @@ def get_context_data(self):
         # Hides the requests and domains buttons in the navbar
         context_stuff["hide_requests"] = self.is_portfolio
         context_stuff["hide_domains"] = self.is_portfolio
+        context_stuff["domain_request_id"] = self.domain_request.id
```
Comment (erinysong) [inline src/registrar/views/domain_request.py:496]: I know this is for pre-existing code not in scope, but thoughts on renaming this variable to `form_context` or something along the lines of that?

**Review by erinysong — COMMENTED** [review]
Verified and tested ACs + code LGTM! Just had two small questions and can approve afterwards

File: src/registrar/views/domain_request.py (hunk 2 above)
Comment (erinysong) [reply src/registrar/views/domain_request.py]: I think just context also works! Glad we were in agreement 🎉

**Review by erinysong — APPROVED** [review]
LGTM! 🚢

**Review by Katherine-Osos — APPROVED** [review]
LGTM!

**Pull request merged**